src/mimefactory.rs

```diff
@@ -384,6 +381,26 @@ impl<'a> MimeFactory<'a> {
         }
     }
 
+    async fn should_attach_selfavatar(context: &Context, msg: &Message) -> bool {
+        let cmd = msg.param.get_cmd();
+        (cmd != SystemMessage::SecurejoinMessage || {
+            let step = msg.param.get(Param::Arg).unwrap_or_default();
+            step == "vg-request-with-auth"
```

Comment:
`vc/vg-auth-required` means that the contact isn't yet verified

Reply:
INVITENUMBER can be just 8 bytes as per the SecureJoin doc. Still difficult to guess if it's random, but the `vc-request` message can be captured and resent by an attacker

Reply:
> Alice sending `vc-auth-required` means that Bob sent correct invite code in `vc/vg-request` message. Invite code is obtained from a QR code which even has an avatar of Alice overlayed in SVG, so Bob has already seen Alice's avatar and display name.

Yes but that doesn't mean Gmail has seen the avatar, and they will if the message is sent unencrypted and worse thing is that it is a leak totally out of user expectations

Reply:
`vc-auth-required` is already encrypted (unlike `vc-request`), but it can be encrypted to an attacker if it captured INVITENUMBER and resent it to Alice. The attacker (e.g. Gmail) can do this only sometimes not to break the QR-join feature completely, but slowly collect users' avatars and other data

Reply:
Added a comment, but i'd prefer to leave the condition as is because if `step` for some reason has the wrong value, i'd also prefer not to send the avatar.